perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,717 @@
+  nms_overlap_thresh_ = model_config["test_cfg"]["nms_thr"][0].as<float>();
+  use_depth_ = model_config["use_depth"].as<bool>();
+  use_adj_ = model_config["use_adj"].as<bool>();
+  transform_size_ = model_config["transform_size"].as<int>();
+  cam_params_size_ = model_config["cam_params_size"].as<int>();
+
+  std::vector<std::vector<float>> nms_factor_temp =
+    model_config["test_cfg"]["nms_rescale_factor"].as<std::vector<std::vector<float>>>();
+  nms_rescale_factor_.clear();
+  for (const auto & task_factors : nms_factor_temp) {
+    for (float factor : task_factors) {
+      nms_rescale_factor_.push_back(factor);
+    }
+  }
+
+  std::vector<std::vector<std::string>> class_name_pre_task;
+  class_num_ = 0;
+  YAML::Node tasks = model_config["model"]["tasks"];
+  class_num_pre_task_ = std::vector<int>();
+  for (auto it : tasks) {
+    int num = it["num_class"].as<int>();
+    class_num_pre_task_.push_back(num);
+    class_num_ += num;
+    class_name_pre_task.push_back(it["class_names"].as<std::vector<std::string>>());
+  }
+
+  YAML::Node common_head_channel = model_config["model"]["common_head"]["channels"];
+  YAML::Node common_head_name = model_config["model"]["common_head"]["names"];
+  for (size_t i = 0; i < common_head_channel.size(); i++) {
+    out_num_task_head_[common_head_name[i].as<std::string>()] = common_head_channel[i].as<int>();
+  }
+
+  feat_h_ = input_img_h_ / down_sample_;
+  feat_w_ = input_img_w_ / down_sample_;
+  depth_num_ = (depth_end_ - depth_start_) / depth_step_;
+  xgrid_num_ = (x_end_ - x_start_) / x_step_;
+  ygrid_num_ = (y_end_ - y_start_) / y_step_;
+  zgrid_num_ = (z_end_ - z_start_) / z_step_;
+  bev_h_ = ygrid_num_;
+  bev_w_ = xgrid_num_;
+
+  post_rot_ << resize_radio_, 0, 0, 0, resize_radio_, 0, 0, 0, 1;
+  post_trans_.translation() << -crop_w_, -crop_h_, 0;
+
+  adj_num_ = 0;
+  if (use_adj_) {
+    adj_num_ = model_config["adj_num"].as<int>();
+  }
+
+  postprocess_ptr_.reset(new PostprocessGPU(
+    class_num_, score_thresh_, nms_overlap_thresh_, nms_pre_maxnum_, nms_post_maxnum_, down_sample_,
+    bev_h_, bev_w_, x_step_, y_step_, x_start_, y_start_, class_num_pre_task_,
+    nms_rescale_factor_));
+}
+
+void BEVDet::mallocDeviceMemory()
+{
+  trt_buffer_sizes_.resize(trt_engine_->getNbBindings());
+  trt_buffer_dev_ = reinterpret_cast<void **>(new void *[trt_engine_->getNbBindings()]);
+  for (int i = 0; i < trt_engine_->getNbBindings(); i++) {
+    nvinfer1::Dims32 dim = trt_context_->getBindingDimensions(i);
+    size_t size = 1;
+    for (int j = 0; j < dim.nbDims; j++) {
+      size *= dim.d[j];
+    }
+    size *= dataTypeToSize(trt_engine_->getBindingDataType(i));
+    trt_buffer_sizes_[i] = size;
+    CHECK_CUDA(cudaMalloc(&trt_buffer_dev_[i], size));
+  }
+
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "img num binding : %d", trt_engine_->getNbBindings());
+
+  post_buffer_ = reinterpret_cast<void **>(new void *[class_num_pre_task_.size() * 6]);
```

Comment:
Similar to the previous suggestion, `post_buffer_` should be a `std::unique_ptr<void*[]>` to better manage its memory.